Dockerfile

```diff
@@ -20,31 +20,12 @@ ENV        NODE_PATH=/calypso/server:/calypso/client
 COPY       ./env-config.sh /tmp/env-config.sh
 RUN        bash /tmp/env-config.sh
 
-# Build a "dependencies" layer
-#
-# This layer should include all required npm modules
-# and should only change as often as the dependencies
-# change. This layer should allow for final build times
-# to be limited only by the Calypso build speed.
-COPY       ./package.json ./npm-shrinkwrap.json /calypso/
-RUN        npm ci --only=production
-
 # Build a "source" layer
 #
 # This layer is populated with up-to-date files from
 # Calypso development.
-#
-# If package.json and npm-shrinkwrap are unchanged,
-# `install-if-deps-outdated` should require no action.
-# However, time is being spent in the build step on
-# `install-if-deps-outdated`. This is because in the
-# following COPY, the npm-shrinkwrap mtime is being
-# updated, which is confusing `install-if-deps-outdated`.
-# Touch after copy to ensure that this layer will
-# not trigger additional install as part of the build
-# in the following step.
 COPY       . /calypso/
-RUN        touch node_modules
+RUN        npm ci --only=production
```

Comment:
@sirreal can you see any way around this? with `file:` refs in the package.json, I don't think we can multi-layer this like we did.

Reply:
with `file:` refs will it matter? those aren't "installed" right - ?

in other words, why does a cached `npm install` which fetches the external libs invalidate a build that also resolves certain modules locally?